.github/workflows/test-wheel-linux.yml

```diff
@@ -149,11 +149,33 @@ jobs:
           SHA: ${{ github.sha }}
         run: ./ci/tools/env-vars test
 
-      - name: Download cuda-pathfinder build artifacts
-        uses: actions/download-artifact@d3f86a106a0bac45b974a628896c90dbdf5c8093  # v4.3.0
-        with:
-          name: cuda-pathfinder-wheel
-          path: ./cuda_pathfinder
+      - name: Download cuda-pathfinder build artifacts from main branch
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # See https://github.com/cli/cli/blob/trunk/docs/install_linux.md#debian-ubuntu-linux-raspberry-pi-os-apt.
+          # gh is needed for artifact fetching.
```

Comment:
Please consolidate/merge the gh-installation step and move it to just before the other steps that need gh. Right now this is repeated code.